COMPARISON/CLUSTERING/GridBasedClustering.h: compiler error

Hello.

`OpenMS-2.2.0` does not compile on fedora with following error:

```
In file included from /builddir/build/BUILD/openms-2.2.0/py2build/src/openms/source/COMPARISON/CLUSTERING/GridBasedClustering.cpp:35:0:
/builddir/build/BUILD/openms-2.2.0/py2build/src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h: In member function 'void OpenMS::GridBasedClustering<Metric>::cluster()':
/builddir/build/BUILD/openms-2.2.0/py2build/src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h:186:52: error: no matching function for call to 'OpenMS::GridBasedClustering<Metric>::startProgress(int, OpenMS::Size&, const char [11])'
       startProgress(0, clusters_start, "clustering");
                                                    ^
In file included from /builddir/build/BUILD/openms-2.2.0/py2build/src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h:40:0,
                 from /builddir/build/BUILD/openms-2.2.0/py2build/src/openms/source/COMPARISON/CLUSTERING/GridBasedClustering.cpp:35:
/builddir/build/BUILD/openms-2.2.0/py2build/src/openms/include/OpenMS/CONCEPT/ProgressLogger.h:112:10: note: candidate: void OpenMS::ProgressLogger::startProgress(OpenMS::SignedSize, OpenMS::SignedSize, const OpenMS::String&) const
     void startProgress(SignedSize begin, SignedSize end, const String& label) const;
          ^~~~~~~~~~~~~
/builddir/build/BUILD/openms-2.2.0/py2build/src/openms/include/OpenMS/CONCEPT/ProgressLogger.h:112:10: note:   no known conversion for argument 3 from 'const char [11]' to 'const OpenMS::String&'
make[3]: *** [src/openms/CMakeFiles/OpenMS.dir/build.make:12153: src/openms/CMakeFiles/OpenMS.dir/source/COMPARISON/CLUSTERING/GridBasedClustering.cpp.o] Error 1
```

Full log: https://kojipkgs.fedoraproject.org//work/tasks/5831/20365831/build.log
Dependencies list: https://kojipkgs.fedoraproject.org//work/tasks/5831/20365831/root.log
Used hardware: https://kojipkgs.fedoraproject.org//work/tasks/5831/20365831/hw_info.log

--- a/src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h
+++ b/src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h
@@ -182,8 +182,10 @@ public:
     void cluster()
     {
       // progress logger
+      // NOTE: for some reason, gcc7 chokes if we remove the OpenMS::String
+      // below, so lets just not change it.
       Size clusters_start = clusters_.size();
-      startProgress(0, clusters_start, "clustering");
+      startProgress(0, clusters_start, OpenMS::String("clustering"));
 
       MinimumDistance zero_distance(-1, -1, 0);
 

--- a/src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h
+++ b/src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h
@@ -37,6 +37,7 @@
 
 #include <OpenMS/KERNEL/StandardTypes.h>
 #include <OpenMS/DATASTRUCTURES/DRange.h>
+#include <OpenMS/DATASTRUCTURES/String.h>
 #include <OpenMS/CONCEPT/ProgressLogger.h>
 #include <OpenMS/COMPARISON/CLUSTERING/ClusteringGrid.h>
 #include <OpenMS/COMPARISON/CLUSTERING/GridBasedCluster.h>

